backend/src/apiserver/storage/db.go

```diff
@@ -88,6 +91,11 @@ func (d MySQLDialect) IsDuplicateError(err error) bool {
 	return ok && sqlError.Number == mysqlerr.ER_DUP_ENTRY
 }
 
+// UpdateFromOrJoin TODO(gkcalat): deprecate resource_references table once we migration to v2beta1 is available.
+func (d MySQLDialect) UpdateWithJointOrFrom(targetTable, joinTable, setClause, joinClause, whereClause string) string {
+	return fmt.Sprintf("UPDATE %s JOIN %s ON %s SET %s WHERE %s", targetTable, joinTable, joinClause, setClause, whereClause)
+}
```

Comment:
Could we limit this function's scope to "UPDATE %s JOIN %s ON %s" and let users set the remaining via `squirrel.SetMap` and `squirrel.Where`? That way, this function will only implement the missing part.

Reply:
NIT: replace `JOIN` with explicit `INNER JOIN` for better readability

Reply:
> Could we limit this function's scope to "UPDATE %s JOIN %s ON %s" and let users set the remaining via `squirrel.SetMap` and `squirrel.Where`? That way, this function will only implement the missing part.

Please, ignore this, according to the discussion below.

One last comment: we recently updated the [code styling requirements](https://github.com/kubeflow/pipelines/blob/master/backend/README.md#code-style). TL;DR; it is recommended to run `pre-commit` with the provided config, which includes a golang linter.

Reply:
Thank you for referencing this. I have just started using pre-commit.